### PR TITLE
Prepare code for new wsdl-reader

### PR DESCRIPTION
**Pull request by veewee**
This PR introduces:

* Engine v2 support https://github.com/php-soap/engine/releases/tag/2.0.0
* Basic configuration to use new wsdl-reader metadata provider (https://github.com/php-soap/wsdl-reader)

### CODE REVIEW

File: src/Phpro/SoapClient/CodeGenerator/Assembler/PropertyAssemblerOptions.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+/**
+ * Class PropertyAssemblerOptions
+ * @package Phpro\SoapClient\CodeGenerator\Assembler
+ */
+class PropertyAssemblerOptions
+{
+    private bool $typeHints = true;
+    private bool $docBlocks = true;
+    private string $visibility = PropertyGenerator::VISIBILITY_PRIVATE;
```
Comment (janvernieuwe) [inline src/Phpro/SoapClient/CodeGenerator/Assembler/PropertyAssemblerOptions.php:16]: No readonly?

File: src/Phpro/SoapClient/Soap/CodeGeneratorEngineFactory.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        string $wsdlLocation,
+        ?WsdlLoader $loader = null,
+        ?MetadataOptions $metadataOptions = null
+    ): Engine {
+        $loader ??= new FlatteningLoader(Psr18Loader::createForClient(Psr18ClientDiscovery::find()));
+        $metadataOptions ??= MetadataOptions::empty()->withTypesManipulator(
+            // Ext-soap is not able to work with duplicate types (see FAQ)
+            // Therefore, we decided to combine all duplicate types into 1 big intersected type by default instead.
+            // Therefore it will always be usable, but might contain some empty properties.
+            // It has it's limitations but it is workable until ext-soap handles XSD namespaces properly.
```
Comment (janvernieuwe) [inline src/Phpro/SoapClient/Soap/CodeGeneratorEngineFactory.php:37]: But this is handled well with the wsdl reader?

File: src/Phpro/SoapClient/CodeGenerator/Assembler/PropertyAssemblerOptions.php (hunk 1 above)
Comment (veewee) [reply src/Phpro/SoapClient/CodeGenerator/Assembler/PropertyAssemblerOptions.php:16]: Currently added this one to make the configuration more flexible.
It is in line with all other option classes.

We could add readonly (in another PR) for the sake of explicitely indicating it, but this class is immutable+readonly by design.

File: test/PhproTest/SoapClient/Unit/CodeGenerator/Model/PropertyTest.php [hunk 3]
```diff
@@ -20,8 +22,8 @@ public function it_returns_null_type_pre_php8(): void
         if (PHP_VERSION_ID >= 80000) {
             self::markTestSkipped('Pre PHP 8 only');
```
[the remark this answers is not in the document]
Comment (veewee) [reply test/PhproTest/SoapClient/Unit/CodeGenerator/Model/PropertyTest.php:23]: Nope, will remove it.

File: src/Phpro/SoapClient/Soap/CodeGeneratorEngineFactory.php (hunk 2 above)
Comment (veewee) [reply src/Phpro/SoapClient/Soap/CodeGeneratorEngineFactory.php:37]: The WSDL reader has all the information it needs to succesfully handle this if we were to made our own encoding / decoding system.
We are still using ext-soap's encoding/decoding internally, so  - sadly - we still have to care about how duplicate types will be handled (removed / merged / ...) for now.

**Pull request merged**